packages/editor/src/store/selectors.js

```diff
@@ -144,12 +144,6 @@ export const isEditedPostDirty = createRegistrySelector(
  */
 export const hasNonPostEntityChanges = createRegistrySelector(
 	( select ) => ( state ) => {
-		const enableFullSiteEditing = getEditorSettings( state )
-			.__experimentalEnableFullSiteEditing;
-		if ( ! enableFullSiteEditing ) {
-			return false;
-		}
```

Comment:
I not certain what this check was used for. Maybe just an optimization?

Reply:
It has a corresponding unit test that fails:
https://github.com/WordPress/gutenberg/pull/26500/checks?check_run_id=1314637813#step:6:6887